build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -164,24 +164,32 @@
                     InputFile="%(_DownloadAndExtractItem.DownloadFileName)"
                     DestinationDirectory="%(_DownloadAndExtractItem.ExtractDestintation)" />
   </Target>
-
-  <Target Name="SetupRestorePackagesInputsOutputs">
-    <ItemGroup>
-      <RestorePackagesInput Include="$(RepoRoot)/src/**/project.json" />
-      <RestorePackagesInput Include="$(RepoRoot)/tools/**/project.json" />
-    </ItemGroup>
-  </Target>
   
   <Target Name="RestorePackages"
           DependsOnTargets="SetupRestorePackagesInputsOutputs;Init"
           Inputs="@(RestorePackagesInput)"
           Outputs="@(RestorePackagesInput->'%(RelativeDir)/project.lock.json')">
 
+    <CallTarget Targets="CleanSrcToolsLockFiles" />
     <DotNetRestore WorkingDirectory="$(RepoRoot)/src" ToolPath="$(DotNetPath)" />
     <DotNetRestore WorkingDirectory="$(RepoRoot)/tools" ToolPath="$(DotNetPath)" />
 
   </Target>
 
+  <Target Name="CleanSrcToolsLockFiles" >
+    <ItemGroup>
+      <SrcToolsLockFiles Include="$(RepoRoot)/src/**/project.lock.json;$(RepoRoot)/tools/**/project.lock.json" />
+    </ItemGroup>
+    <Delete Files="@(SrcToolsLockFiles)" /> 
```

Comment:
This comment would end up in 5 places and I don't think it's necessary.

We are using a very brute-force incrementality algorithm. If ANY project.json is newer than ANY associated lock file then do restore. NuGet, only updates lock files if they are different than before to prevent restore from breaking pj-style incrementality [which is the right thing to do]. This means, however, that updating a single PJ makes it permanently newer than all lock files for any other PJ in the search path. This in turn means that incrementality is broken until all of those lock files are deleted and replaced. This target, then, is invoked if we will restore under a given root to ensure that that restore replaces ALL lock files and therefore re-enables incrementality.